TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementBuilder.cs

```diff
@@ -49,19 +52,73 @@ private SelectClause CreateSelectClause(CodeElementsParser.Sql_selectClauseConte
+                    ParseTreeUtils.GetFirstToken(context.SQL_DISTINCT()));
             }
             if (context.star() != null)
             {
                 return new SelectClause(selectionModifier, new StarSelection());
             }
 
+            if (context.selections() != null)
+            {
+                var dotStarSelections = new List<DotStarSelection>();
+                int l = context.selections().ChildCount;
+                int i = 0;
+                foreach (var selection in context.selections().selection())
+                {
+                    dotStarSelections.Add(CreateDotStarSelection(selection));
+                }
+                return new SelectClause(selectionModifier, dotStarSelections);
+            }
+
             return null;
+        }
+
+        public DotStarSelection CreateDotStarSelection(CodeElementsParser.SelectionContext context)
+        {
+            if (context.dotStarSelection() != null)
+            {
+                if ((context.dotStarSelection().tableOrViewOrCorrelationName().Name != null))
+                {
+                    Token name = context.dotStarSelection().tableOrViewOrCorrelationName().Name as Token;
+                    SymbolReference tail = new SymbolReference(new AlphanumericValue(name), SymbolType.AlphabetName);
+                    if (context.dotStarSelection().tableOrViewOrCorrelationName().SchemaName != null)
+                    {
+                        Token schemaName =
+                            context.dotStarSelection().tableOrViewOrCorrelationName().SchemaName as Token;
+                        SymbolReference SchemaName =
+                            new SymbolReference(new AlphanumericValue(schemaName), SymbolType.AlphabetName);
+                        if (context.dotStarSelection().tableOrViewOrCorrelationName().DBMS != null)
+                        {
+                            Token dbms =
+                                context.dotStarSelection().tableOrViewOrCorrelationName().DBMS as Token;
+                            SymbolReference DBMS =
+                                new SymbolReference(new AlphanumericValue(dbms), SymbolType.AlphabetName);
+                            QualifiedSymbolReference head = new QualifiedSymbolReference(DBMS, SchemaName);
+                            QualifiedSymbolReference fullName = new QualifiedSymbolReference(head, tail);
+                            return new DotStarSelection(fullName);
+                        }
+                        else
+                        {
+                            QualifiedSymbolReference fullName = new QualifiedSymbolReference(SchemaName, tail);
+                            return new DotStarSelection(fullName);
+                        }
+                    }
+                    else
+                    {
+                        return new DotStarSelection(tail);
+                    }
```

Comment:
I think we should refactor this as follows:
- extract all three parts of the name into Token instances, the first 2 may be null
- call a method to create the appropriate SymbolReference given the three tokens. This new method will be reusable for column names for example

As for the `SymbolType`, this is not corect. `AlphabetName` is a Cobol concept and has no meaning in SQL. We have to create new symbol types. So add a `SqlTableOrViewOrCorrelationName` member in the enum and use it.